processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -363,34 +386,47 @@ static double estimateSelectivityOfBitmapTree(
    * <p>
    * The non-overlap ratio can be computed like below.
    * <p>
-   * nonOverlapRatio(b1, b2) = size(union(b1, b2)) / (size(b1) + size(b2))
+   * overlapSize = size(b1) + size(b2) - size(union(b1, b2))
+   * nonOverlapRatio(b2) = (size(b2) - overlapSize) / size(b2)
+   * <p>
+   * The approximate unioned size is
+   * <p>
+   * unionedSize = size(b1) + size(b2) * nonOverlapRatio(b2)
    * <p>
    * Given bitmaps, this method calculates the non-overlap ratios of N bitmap samples,
    * and then returns the average of them.
    *
    * @param bitmapIndex   bitmap index to retrieve bitmaps
    * @param bitmapIndexes a list of indexes of bitmaps
    *
-   * @return approximated average non-overlap ratio of bitmaps
+   * @return approximated average non-overlap ratio of bitmaps.
    *
    * @see #estimatePredicateSelectivity(ColumnSelector, String, BitmapIndexSelector, Predicate)
    */
-  private static double computeNonOverlapRatioFromRandomBitmapSamples(
+  static double computeNonOverlapRatioFromRandomBitmapSamples(
```

Comment:
I don't believe this model as well. It computes mean of values, which are position-dependent (discount in time). Consider you have two bitmaps, covers 10% of bits, another 100%. If you select them in one order you will get 0.0 from this method, in another order - 0.9.

Another problem with this estimation, is that is computes pairwise unions, i. e. generates a lot of garbage.

I suggest to not try compute selectivity with multi-value dimensions in this PR.

Later, I propose the following approach to be implemented:
 - Compute a sample union of N randomly selected bitmaps (not one-by-one, union of all of them together)
 - Based on 1) size of the sample union 2) sizes of all unified bitmaps 3) total number of rows 4) number and sizes of bitmaps not included in the sample union, compute the expected size of union of all bitmaps. A non-trivial math model is required to do this, and I don't expect it could be done without help or at least review by somebody who remembers statistics course from high school really well.
 - Reuse the computed sample union later in Filter/UseIndexesStrategy, when the union of all bitmaps is needed (and if needed).

Also for efficiency of the union size estimation, some parameters could be pre-computed when segment is created and stored in the segment format. For example, the sum of sizes of all bitmaps (which is bigger than the number of rows, for multi-value dimensions). This is related to #3882.